### PR TITLE
Be resilient to when we get an exception while reading metadata.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jasonmalinowski @basoundr  @dotnet/roslyn-ide 

Thanks!

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
Tagging @tmat For the compiler documentation change.

File: src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs [hunk 1]
```diff
@@ -13,6 +14,18 @@ namespace Microsoft.CodeAnalysis.FindSymbols
 {
     internal partial class SymbolTreeInfo
     {
+        private static Metadata GetMetadataNoThrow(PortableExecutableReference reference)
+        {
+            try
+            {
+                return reference.GetMetadata();
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs]: Should this just wrap in PerformIO?

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by basoundr** [comment]
:+1:

File: src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs (hunk 1 above)
Comment (tmat) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs]: No, that wouldn't handle BadImageFormatException. The method documents exceptions that need to be handled.

**Pull request merged**